commit/merkleroot/outcome.go

```diff
@@ -138,15 +139,34 @@ func buildReport(
 	sort.Slice(roots, func(i, j int) bool { return roots[i].ChainSel < roots[j].ChainSel })
 
 	sigs := make([]cciptypes.RMNECDSASignature, 0)
-	if q.RMNSignatures == nil {
-		lggr.Warn("RMNSignatures are nil")
-	} else {
+	if q.RMNSignatures != nil { // TODO: should never be nil, error after e2e RMN integration.
 		parsedSigs, err := rmn.NewECDSASigsFromPB(q.RMNSignatures.Signatures)
 		if err != nil {
-			lggr.Errorw("Failed to parse RMN signatures", "error", err)
-		} else {
-			sigs = parsedSigs
+			lggr.Errorw("Failed to parse RMN signatures returning an empty outcome", "error", err)
+			return Outcome{}
 		}
+		sigs = parsedSigs
+
+		signedRoots := mapset.NewSet[cciptypes.MerkleRootChain]()
+		for _, laneUpdate := range q.RMNSignatures.LaneUpdates {
```

Comment:
I'm not sure if we do these validations elsewhere, but one thing I can think of is do we validate that RMN nodes don't send overlapping intervals / different roots for same interval / etc. ?

Reply:
Yes we handle all this